File: Python/applications/number-guess.py
import random
from enum import Enum

class NumberGuess():
    class HighLow(Enum):
        HIGH = 1
        WIN = 0
        LOW = -1
        LOST = -2

    def __init__(self):
        self.number = random.randrange(0, 100)
        self.guesses = 10
        self.string_values = {
            self.HighLow.HIGH : "Too High!",
            self.HighLow.WIN : "You win! The number was {number}".format(number=self.number),
            self.HighLow.LOW : "Too Low!",
            self.HighLow.LOST : "You loose! The number was {number}".format(number=self.number)
        }

    def parse_guess(self, guess: int) -> int:
        if guess > self.number:
            return self.HighLow.HIGH
        elif guess < self.number:
            return self.HighLow.LOW

        return self.HighLow.WIN

    def make_guess(self, guess):
        self.guesses -= 1
        if self.guesses == 0:
            return self.HighLow.LOST
        return self.parse_guess(guess)

    def game_loop(self):
        while self.guesses != 0:
            guess = int(input("Make your guess, you have {guesses} left: ".format(guesses=self.guesses)))
            result = self.make_guess(guess)
            if result == self.HighLow.WIN or result == self.HighLow.LOST:
                return self.string_values[result]
            print(self.string_values[result])

if __name__ == '__main__':
    game = NumberGuess()
    print(game.game_loop())

File: Java/number-guess/src/main/java/NumberGuess.java
import java.util.HashMap;
import java.util.Map;
import java.util.Random;
import java.util.Scanner;

public class NumberGuess {
    private int number;
    private int guesses;
    private static Map<HighLow, String> stringValues = new HashMap<>();

    private Random random;
    private Scanner scanner;

    enum HighLow {
        HIGH,
        WIN,
        LOW,
        LOST
    }

    public NumberGuess() {
        this.random = new Random();
        this.scanner = new Scanner(System.in);

        this.number = random.nextInt(101);
        this.guesses = 10;

        System.out.println(this.number);

        stringValues.put(HighLow.HIGH, "Too High!");
        stringValues.put(HighLow.WIN, String.format("You win! The number was %d", this.number));
        stringValues.put(HighLow.LOW, "Too Low!");
        stringValues.put(HighLow.LOST, String.format("You loose! The number was %d", this.number));
    }

    private HighLow parse_guess(int guess) {
        if (guess > this.number) {
            return HighLow.HIGH;
        } else if (guess < this.number) {
            return HighLow.LOW;
        }
        return HighLow.WIN;
    }

    private HighLow makeGuess(int guess) {
        this.guesses -= 1;
        if (this.guesses == 0) {
            return HighLow.LOST;
        }
        return parse_guess(guess);
    }

    private int getUserInput() {
        System.out.println(String.format("Make your guess, you have %d guesses left: ", this.guesses));
        return scanner.nextInt();
    }

    public String gameLoop() {
        while(this.guesses != 0) {
            int guess = getUserInput();
            HighLow result = makeGuess(guess);
            if (result == HighLow.WIN || result == HighLow.LOST) {
                return stringValues.get(result);
            }
            System.out.println(stringValues.get(result));
        }
        return "";
    }
}
